kolibri/plugins/learn/assets/src/views/LibraryPage.vue

```diff
@@ -1,103 +1,208 @@
-            :contents="trimmedPopular"
-            @toggleInfoPanel="toggleInfoPanel"
+          <KIconButton
+            icon="channel"
+            :ariaLabel="$tr('viewAsGrid')"
+            :color="$themeTokens.text"
+            :tooltip="$tr('viewAsGrid')"
+            @click="toggleCardView('card')"
           />
         </div>
-        <div v-else>
-          <KCircularLoader
-            v-if="searchLoading"
-            class="loader"
-            type="indeterminate"
-            :delay="false"
+        <h2>{{ $tr('recent') }}</h2>
+        <HybridLearningCardGrid
+          v-if="popular.length"
+          :cardViewStyle="currentViewStyle"
+          :numCols="numCols"
+          :genContentLink="genContentLink"
+          :contents="trimmedPopular"
+          :currentPage="currentPage"
+        />
+      </div>
+
+      <!-- Display of search results, after the search is finished loading -->
+
+      <!-- First section is the results title and the various display buttons  -->
+      <!-- for interacting or updating the results   -->
+      <div v-else-if="!searchLoading" class="results-title">
+        <h2 class="results-title">
+          {{ $tr('results', { results: results.length }) }}
+        </h2>
+        <KButton
+          v-if="more"
+          :text="coreString('viewMoreAction')"
+          appearance="basic-link"
+          :disabled="moreLoading"
+          class="filter-action-button"
+          @click="searchMore"
+        />
+        <div v-if="!(windowBreakpoint < 1)" class="toggle-view-buttons">
+          <KIconButton
+            icon="menu"
+            :ariaLabel="$tr('viewAsList')"
+            :color="$themeTokens.text"
+            :tooltip="$tr('viewAsList')"
+            @click="toggleCardView('list')"
+          />
+          <KIconButton
+            icon="channel"
+            :ariaLabel="$tr('viewAsGrid')"
+            :color="$themeTokens.text"
+            :tooltip="$tr('viewAsGrid')"
+            @click="toggleCardView('card')"
           />
-          <div v-else>
-            <h2>{{ $tr('results', { results: results.length }) }}</h2>
-            <KButton
-              v-if="more"
-              :text="coreString('viewMoreAction')"
-              :primary="false"
-              :disabled="moreLoading"
-              @click="searchMore"
-            />
-            <p>{{ $tr('clearAll') }}</p>
-            <ContentCardGroupGrid
-              v-if="results.length"
-              :cardViewStyle="currentViewStyle"
-              :genContentLink="genContentLink"
-              :contents="results"
-            />
-            <KButton
-              v-if="more"
-              :text="coreString('viewMoreAction')"
-              :primary="false"
-              :disabled="moreLoading"
-              @click="searchMore"
-            />
+        </div>
+        <div class="results-header-group">
+          <div
+            v-for="item in Object.values(searchTermChipList)"
```

Comment:
To avoid having to do the search in `removeFilterTag` you can preserve both the key and the value here.
```suggestion
            v-for="(item, key) in searchTermChipList"
```